distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Header.enso

```diff
@@ -182,3 +167,13 @@ type Header
              example_header_text_plain = Header.text_plain
     text_plain : Header
     text_plain = Header.content_type "text/plain"
+
+## PRIVATE
+type Header_Comparator
+    compare x y =
+        if x.name.equals_ignore_case y.name && x.value == y.value then Ordering.Equal else
+            Nothing
+
+    hash x = Comparable.hash_builtin [x.name.to_case_insensitive_key, x.value]
```

Comment:
I am not particularly happy about direct usage of `Comparable.hash_builtin` here. Although std lib is a special library that might access some internal parts of the engine, we should strive to limit that as much as possible. In this case, however, I understand this workaround. There is unfortunately, no direct way how to get a hash code of a vector. Since `x.name` and `x.value` should be `Text`, if I am not mistaken, I would rather suggest doing something like this:
```
hash x =
    comp = Comparable.from x.name
    comp.hash (x.name.to_case_insensitive_key + x.value)
```